timesketch/lib/tasks.py

```diff
@@ -181,37 +186,78 @@ def build_index_pipeline(file_path, timeline_name, index_name, file_extension,
     return chain(index_task, index_analyzer_chain)
 
 
-def build_sketch_analysis_pipeline(sketch_id):
+def build_sketch_analysis_pipeline(sketch_id, searchindex_id, user_id,
+                                   analyzer_names=None, analyzer_kwargs=None):
     """Build a pipeline for sketch analysis.
 
     Args:
-        sketch_id: The ID of the sketch to analyze.
+        sketch_id (int): The ID of the sketch to analyze.
+        searchindex_id (int): The ID of the searchindex to analyze.
+        user_id (int): The ID of the user who started the analyzer.
+        analyzer_names (list): List of analyzers to run.
+        analyzer_kwargs (dict): Arguments to the analyzers.
 
     Returns:
         Celery group with analysis tasks or None if no analyzers are enabled.
     """
     tasks = []
+    auto_analyzers = current_app.config.get('AUTO_SKETCH_ANALYZERS', None)
 
-    # Exit early if sketch analyzers are disabled.
-    if not current_app.config.get('ENABLE_SKETCH_ANALYZERS', False):
+    # Exit early if no sketch analyzers are configured to run.
+    if not (analyzer_names or auto_analyzers):
         return None
 
-    for analyzer_name, analyzer_cls in manager.AnalysisManager.get_analyzers():
-        kwarg_list = analyzer_cls.get_kwargs()
+    if not analyzer_names:
+        analyzer_names = auto_analyzers
+
+    if user_id:
+        user = User.query.get(user_id)
+    else:
+        user = None
 
+    sketch = Sketch.query.get(sketch_id)
+    analysis_session = AnalysisSession(user, sketch)
+
+    analyzers = manager.AnalysisManager.get_analyzers(analyzer_names)
+    for analyzer_name, analyzer_cls in analyzers:
         if not analyzer_cls.IS_SKETCH_ANALYZER:
             continue
 
-        if kwarg_list:
-            for kwargs in kwarg_list:
-                tasks.append(run_sketch_analyzer.s(
-                    sketch_id, analyzer_name, **kwargs))
+        if analyzer_kwargs:
```

Comment:
what are you doing when it comes to auto analyzers? some of them might require some kwargs? how are you passing them along?

Reply:
I was planning on adding TODOs for this, but missed adding that. Good catch! I decided to fix this now instead as this is a small fix. You will be able to pass in kwargs to auto analyzers via config, and for any other manually run analyzer there will be a config option to specify default kwargs per analyzer as well. Explicitly provided kwargs will always take precedence though.
```
AUTO_SKETCH_ANALYZERS_KWARGS = {'domain': {'foo': 'bar'}} 
ANALYZERS_DEFAULT_KWARGS = {'domain': {'foo': 'bar'}} 
```